### PR TITLE
Added binary sparsemax loss and corresponding test.

### PR DESCRIPTION
**Pull request by vroulet**
Issue #35

### CODE REVIEW

File: tests/loss_test.py [hunk 1]
```diff
@@ -92,6 +92,19 @@ def test_binary_logistic_loss(self):
...
 
+  def test_binary_sparsemax_loss(self):
+    # @vroulet: There is no reference implementation for this loss as far as I know.
+    # Maybe it would be better to allow no reference implementation in
+    # _test_binary_loss_function
+    def reference_impl(label, logit):
+      scores = jnp.where(label, -logit, logit)
+      return jnp.where(scores < -1.0, 0.0, jnp.where(scores > 1.0, scores, (scores + 1.0)**2/4))
+
+    def invlink(logit):
```
Comment (mblondel) [inline tests/loss_test.py:103]: We can call it `sparse_sigmoid`. Like the usual sigmoid, we have `sparse_sigmoid(0) = 0.5`.

**Review by mblondel — COMMENTED** [review]
Thanks @vroulet for the contribution and good job! I left some comments below. I think we should introduce two new public functions `jaxopt.loss.sparseplus` and `jaxopt.loss.sparse_sigmoid`.

File: jaxopt/_src/loss.py [hunk 2]
```diff
@@ -64,6 +64,23 @@ def binary_logistic_loss(label: int, logit: float) -> float:
...
+    logit: score produced by the model (float).
+  Returns:
+    loss value
+
+  References:
+    Learning with Fenchel-Young Losses. Mathieu Blondel, André F. T. Martins, 
+    Vlad Niculae. JMLR 2020. (Sec. 4.4)
+  """
+  scores = jnp.where(label, -logit, logit)
+  return jnp.where(scores < -1.0, 0.0, jnp.where(scores > 1.0, scores, (scores + 1.0)**2/4))
```
[the remark this answers is not in the document]
Comment (vroulet) [reply jaxopt/_src/loss.py]: Good point! I added it.

File: tests/loss_test.py [hunk 3]
```diff
@@ -92,6 +92,19 @@ def test_binary_logistic_loss(self):
     self._test_binary_loss_function(loss.binary_logistic_loss, sigmoid,
                                     reference_impl)
 
+  def test_binary_sparsemax_loss(self):
+    # @vroulet: There is no reference implementation for this loss as far as I know.
+    # Maybe it would be better to allow no reference implementation in
+    # _test_binary_loss_function
```
[the remark this answers is not in the document]
Comment (vroulet) [reply tests/loss_test.py]: Good idea, I did it.

File: tests/loss_test.py (hunk 1 above)
Comment (vroulet) [reply tests/loss_test.py]: I added it too!

**Review by mblondel — COMMENTED** [review]
Thanks a lot! LGTM. The only thing remaining is the documentation. If you could add the new functions in

https://github.com/google/jaxopt/blob/main/docs/api.rst

and in 

https://github.com/google/jaxopt/blob/main/docs/objective_and_loss.rst

it would be great. Also, could you squash the commits? (1 PR = 1 commit)

**Review by mblondel — APPROVED** [review]
